### PR TITLE
model: Implement FloatHistogram.Compact

### PR DESCRIPTION
**Pull request by beorn7**
__As usual, nothing to see here if you are not into sparsehistogram development.__

@codesome my last contribution for this year. :o)

As anticipated, it's fiddly code, but it seems to work..

### CODE REVIEW

File: model/histogram/float_histogram.go [hunk 1]
```diff
@@ -305,14 +305,164 @@ func addBucket(
...
+		}
+		spans = spans[:iSpan+1]
+		iSpan = 0
+	}
+
+	// Merge spans with zero-length to avoid special cases later.
+	for i, span := range spans {
+		if span.Length == 0 {
+			if iSpan+1 < len(spans) {
+				spans[iSpan+1].Offset += span.Offset
```
Comment (codesome) [inline model/histogram/float_histogram.go:361]: Should it be `i+1` instead of `iSpan+1`?

Comment (codesome) [reply model/histogram/float_histogram.go:361]: If yes, then it needs an unit test.

**Review by codesome — COMMENTED** [review]
Other than the above comment, the implementation LGTM

File: model/histogram/float_histogram.go (hunk 1 above)
Comment (codesome) [reply model/histogram/float_histogram.go:361]: Confirmed that it should be `i+1` with this test case
```go
		{
			"eliminate zero length 2",
			&FloatHistogram{
				PositiveSpans:   []Span{{-2, 2}, {2, 0}, {2, 0}, {2, 0}, {3, 3}},
				PositiveBuckets: []float64{1, 3.3, 4.2, 0.1, 3.3},
			},
			0,
			&FloatHistogram{
				PositiveSpans:   []Span{{-2, 2}, {9, 3}},
				PositiveBuckets: []float64{1, 3.3, 4.2, 0.1, 3.3},
			},
		},
```

Comment (beorn7) [reply model/histogram/float_histogram.go:361]: Great catch. Added fix and test.

**Pull request merged**